### PR TITLE
chore: Add `checkbox` example

### PR DESCRIPTION
**Pull request by AlexMunoz**
Add checkbox example #939

### CODE REVIEW

File: packages/ariakit/src/checkbox/__examples__/checkbox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Checkbox } from "ariakit/checkbox";
+import "./style.css";
+
+export default function Example() {
+  return (
+    <label className="label">
+      <Checkbox className="checkbox" />
+      Checkbox
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox/index.tsx]: Let's use a more real-world example. Maybe this?
```suggestion
      I agree to the terms and conditions
```

File: packages/ariakit/src/checkbox/__examples__/checkbox/style.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.label {
+  @apply flex flex-row items-center gap-2;
+}
+
+.checkbox {
+  @apply ariakit-focus-visible:ariakit-outline;
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox/style.css:6]: Nice! 👍

File: packages/ariakit/src/checkbox/__examples__/checkbox/test.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  getByLabelText,
+  getByRole,
+  press,
+  render,
+} from "ariakit-test-utils";
+import Example from ".";
+
+test("render checkbox", async () => {
+  render(<Example />);
+  expect(getByRole("checkbox")).toMatchInlineSnapshot(`
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox/test.tsx]: Maybe we should get the wrapping label here to get the whole markup. Does `getByLabelText()` return the label element as well?

**Review by diegohaz — COMMENTED** [review]
Nice @AlexMunoz! Made a few suggestions.

File: packages/ariakit/src/checkbox/__examples__/checkbox/test.tsx (hunk 3 above)
Comment (AlexMunoz) [reply packages/ariakit/src/checkbox/__examples__/checkbox/test.tsx]: getByLabelText  returns only the input field. The alternative would be to make a snapshot using the container element.

File: packages/ariakit/src/checkbox/__examples__/checkbox/index.tsx (hunk 1 above)
Comment (AlexMunoz) [reply packages/ariakit/src/checkbox/__examples__/checkbox/index.tsx]: i updated this.

**Review by diegohaz — APPROVED** [review]
Awesome! Thanks!

**Pull request merged**